libs/community/tests/unit_tests/document_loaders/blob_loaders/test_cloud_blob_loader.py

```diff
@@ -0,0 +1,166 @@
+"""Verify that file system blob loader works as expected."""
+import os
+import tempfile
+from typing import Generator
+from urllib.parse import urlparse
+
+import pytest
+
+from langchain_community.document_loaders.blob_loaders import CloudBlobLoader
+
+
+@pytest.fixture
+def toy_dir() -> Generator[str, None, None]:
+    """Yield a pre-populated directory to test the blob loader."""
+    with tempfile.TemporaryDirectory() as temp_dir:
+        # Create test.txt
+        with open(os.path.join(temp_dir, "test.txt"), "w") as test_txt:
+            test_txt.write("This is a test.txt file.")
+
+        # Create test.html
+        with open(os.path.join(temp_dir, "test.html"), "w") as test_html:
+            test_html.write(
+                "<html><body><h1>This is a test.html file.</h1></body></html>"
+            )
+
+        # Create .hidden_file
+        with open(os.path.join(temp_dir, ".hidden_file"), "w") as hidden_file:
+            hidden_file.write("This is a hidden file.")
+
+        # Create some_dir/nested_file.txt
+        some_dir = os.path.join(temp_dir, "some_dir")
+        os.makedirs(some_dir)
+        with open(os.path.join(some_dir, "nested_file.txt"), "w") as nested_file:
+            nested_file.write("This is a nested_file.txt file.")
+
+        # Create some_dir/other_dir/more_nested.txt
+        other_dir = os.path.join(some_dir, "other_dir")
+        os.makedirs(other_dir)
+        with open(os.path.join(other_dir, "more_nested.txt"), "w") as nested_file:
+            nested_file.write("This is a more_nested.txt file.")
+
+        yield f"file://{temp_dir}"
```

Comment:
**Issue**: The 'toy_dir' fixture uses a 'file://' prefix for the directory path, which might not be compatible with all blob storage interfaces or misuse in tests where URL schemes are crucial. <br> **Fix**: Modify the 'toy_dir' fixture to ensure compatibility across different blob storage interfaces by removing the 'file://' prefix when yielding the directory path. <br> **Code Suggestion**: 
 ```
 @@ -42,1 +42,1 @@
 -        yield f"file://{temp_dir}"
 +        yield temp_dir
 ```